test/FIFOMarket.test.ts

```diff
@@ -530,6 +526,99 @@ describe('FIFOMarket', () => {
       );
     });
   });
+
+  describe('nextRemovalForSale', () => {
+    it('should correctly report the next removal for sale if there is inventory', async () => {
+      const buyerInitialBPNoriBalance = formatTokenAmount(1_000_000);
+      const { removal, fifoMarket, hre } = await setupTestLocal({
+        buyerInitialBPNoriBalance,
+      });
+      const { supplier } = hre.namedAccounts;
+      const totalAvailableSupply = '100';
+
+      const list = true;
+      const packedData = hre.ethers.utils.defaultAbiCoder.encode(
+        ['address', 'bool'],
+        [fifoMarket.address, list]
+      );
+      const removalId = await createRemovalTokenId(removal, {
+        supplierAddress: supplier,
+      });
+
+      await Promise.all([
+        removal.mintBatch(
+          supplier,
+          [hre.ethers.utils.parseUnits(totalAvailableSupply)],
+          [removalId],
+          packedData
+        ),
+      ]);
+
+      expect(await fifoMarket.nextRemovalForSale()).to.equal(removalId);
+    });
+    it('should correctly report the next removal for sale if there are multiple removals and some were purchased', async () => {
+      const buyerInitialBPNoriBalance = formatTokenAmount(1_000_000);
+      const { bpNori, removal, fifoMarket, hre } = await setupTestLocal({
+        buyerInitialBPNoriBalance,
+      });
+      const { supplier, buyer } = hre.namedAccounts;
+      const tokenIds = await Promise.all([
+        createRemovalTokenId(removal, {
+          supplierAddress: supplier,
+          vintage: 2018,
+        }),
+        createRemovalTokenId(removal, {
+          supplierAddress: supplier,
+          vintage: 2019,
+        }),
+        createRemovalTokenId(removal, {
+          supplierAddress: supplier,
+          vintage: 2020,
+        }),
+      ]);
+      const removalBalance1 = '5';
+      const removalBalance2 = '5';
+      const removalBalance3 = '5';
```

Comment:
we should just store these as bignumbers here and then you can `.add` below  for total price and not need to `parseUnits` below that in `mintBatch`

Reply:
gonna leave this per our discussion because it's fiddly either way